openedx/core/djangoapps/lang_pref/middleware.py

```diff
@@ -67,15 +67,16 @@ def process_response(self, request, response):
                     # If we can't find the user preferences, then don't modify the cookie
                     pass
 
-            # If set, set the user_pref in the LANGUAGE_COOKIE
-            if user_pref:
+            # If set and user is not a masquerading user, set the user_pref in the LANGUAGE_COOKIE
```

Comment:
I'm wondering whether it would be simpler and maybe better to use the real user's language preference, and to not have the staff user switch languages to that of the learner they are masquerading as. I imagine it could be very difficult if the staff user doesn't speak the language that they are switched over to.

You could just do something as simple as:

   user = request.user.real_user or request.user